Packages/com.chisel.graph/Chisel/ChiselGraphInstance.cs

```diff
@@ -0,0 +1,131 @@
+using Chisel.Core;
+using System;
+using System.Collections.Generic;
+using System.Reflection;
+using Unity.Collections;
+using Unity.Jobs;
+using UnityEngine;
+
+namespace Chisel.Nodes
+{
+    [ExecuteInEditMode]
+    public class ChiselGraphInstance : MonoBehaviour
+    {
+        public ChiselGraph graph;
+        public MeshFilter meshFilter;
+        public MeshRenderer meshRenderer;
+        public bool IsDirty { get; set; }
+
+        public List<GraphProperty> properties;
+        public Dictionary<string, GraphProperty> overriddenProperties;
+
+        CSGTree tree;
+        List<Mesh> meshes;
+
+        void Start()
+        {
+            meshes = new List<Mesh>();
+            UpdateCSG();
+        }
+
+        void Update()
+        {
+            UpdateCSG();
+        }
+
+        void OnValidate()
+        {
+            graph.instance = this;
+
+            if (properties == null || graph.properties.Count != properties.Count)
+                InitProperties();
+
+            for (int i = 0; i < graph.properties.Count; i++)
+                if (graph.properties[i].Name != properties[i].Name)
+                    InitProperties();
```

Comment:
OnValidate is called a lot in Unity, and string comparisons are slow. This loop will get slower and slower the more complicated the graph is, so it makes sense to find another way of doing this that scales better

Reply:
Will hashset be better? btw anyway to sync the properties without destroying existing properties value?

Reply:
Is it possible to set a flag when a name changes instead? Or at least limit this to when editing the graph itself somehow?